File: CS/SubReport.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace Rep3
{
    public partial class SubReport : DevExpress.XtraReports.UI.XtraReport
    {
        public SubReport()
        {
            InitializeComponent();
        }

        int catid;

        public int CategoryID
        {
            get { return catid; }
            set { catid= value; }
        }

        private void SubReport_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e)
        {
            this.productsTableAdapter.FillBy(nwindDataSet1.Products, (int?)CategoryID);
        }

    }
}

File: VB/SubReport.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace Rep3
	Partial Public Class SubReport
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private catid As Integer

		Public Property CategoryID() As Integer
			Get
				Return catid
			End Get
			Set(ByVal value As Integer)
				catid= value
			End Set
		End Property

		Private Sub SubReport_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles MyBase.BeforePrint
			Me.productsTableAdapter.FillBy(nwindDataSet1.Products, CType(CategoryID, Integer?))
		End Sub

	End Class
End Namespace
